### DISCUSSION TITLE
How do I find the config file

### DISCUSSION THREAD

**Discussion by Adaptz** [JanDeDobbeleer/oh-my-posh#1934] [Troubleshoot]
I cannot find where the config file is I really don't understand how this works the set theme just makes the theme temporarily and the website did not help.

**Comment by JanDeDobbeleer** [comment]
When using the guide, we reference a remote config file to get up and running quickly. The config file is cached, but it's also overriding this every time you start you shell. This means any alterations are also reverted. The path can be found using `oh-my-posh config get cache-path`.

If you want to tweak the file, it's advised to copy that to a location of choice (usually your user's home folder) and initialize oh-my-posh with that config file using the path (e.g. `oh-my-posh prompt init pwsh --config ~/mytheme.omp.json | Invoke-Expression`)

I agree the docs can specify this a lot better in the customization section so I'll iterate on that!

**Reply by nevernotmove** [reply]
`oh-my-posh config get cache-path` just gives me `1675517031273`. Is there another way to get the config path?

**Reply by JanDeDobbeleer** [reply]
@nevernotmove wait, the config is the file you reference using the `--config` flag. If you didn't do that, you have the default, embedded config which you can export to a location and format of choice using `oh-my-posh config export --output ~/.mytheme.omp.json`

See [here](https://ohmyposh.dev/docs/installation/customize).

**Reply by nevernotmove** [reply]
@JanDeDobbeleer Thank you for the clarification!

**Reply by mpstaton** [reply]
mpstaton@m4 .config % oh-my-posh prompt init pwsh --config oh-my-posh/config.json
(@(& '/opt/homebrew/bin/oh-my-posh' init pwsh --config='/Users/mpstaton/.config/oh-my-posh/config.json' --print) -join "`n") | Invoke-Expression%                                                     mpstaton@m4 .config % oh-my-posh prompt init pwsh --config oh-my-posh/config.json | Invoke-Expression
zsh: command not found: Invoke-Expression

**Comment by mpstaton** [comment]
> I cannot find where the config file is I really don't understand how this works the set theme just makes the theme temporarily and the website did not help.

I know where the config file is I just don't know how to get it to become the default.  

`mpstaton@m4 .config % oh-my-posh prompt init pwsh --config oh-my-posh/config.json --shell uni
Error: unknown flag: --shell
Usage:
  oh-my-posh prompt init [bash|zsh|fish|powershell|pwsh|cmd|nu|tcsh|elvish|xonsh] [flags]

Flags:
      --debug    enable/disable debug mode
  -h, --help     help for init
  -p, --print    print the init script
  -s, --strict   run in strict mode

Global Flags:
  -c, --config string   config file path`

**Reply by JanDeDobbeleer** [reply]
@mpstaton you're using a deprecated command. `--shell` is not needed, the syntax is `oh-my-posh init pwsh` (or another) as indicated in the [docs](https://ohmyposh.dev/docs/installation/customize#set-the-configuration).
